components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent') -- Note: This can be overwritten
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local WidgetInjector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+-- local LpdbInjector = Lua.import('Module...', {requireDevIfEnabled = true})
+
+local PrizePool = Class.new(function(self, ...) self:init(...) end)
+
+local TODAY = os.date('%Y-%m-%d')
+
+-- TODO: Extract Enum handling to its own Module
+local prizeTypes = Table.map({
+	'USD', 'LOCAL_CURRENCY', 'POINTS', 'QUALIFIES', 'FREETEXT'
+}, function(value, name) return name, value end)
+
+local specialDataTypes = Table.map({
+	'LASTVS', 'GROUPSCORE', 'LASTVSSCORE'
+}, function(value, name) return name, value end)
+
+local prizeData = {
+	[prizeTypes.USD] = {
+		row = 'usdprize',
+		rowParse = function (prizePool, input, context, index)
+			prizePool.hasUSD = true
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.LOCAL_CURRENCY] = {
+		header = 'localcurrency',
+		headerParse = function (prizePool, input, context, index)
+			return {currency = string.upper(input)}
+		end,
+		row = 'localprize',
+		rowParse = function (prizePool, input, context, index)
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.QUALIFIES] = {
+		header = 'qualifies',
+		headerParse = function (prizePool, input, context, index)
+			local link = input:gsub(' ', '_')
+			local data = {link = link}
+
+			-- Automatically retrieve information from the Tournament
+			local tournamentData = prizePool:_getTournamentInfo(link)
+			if tournamentData then
+				data.title = tournamentData.tickername
+				data.icon = tournamentData.icon
+				data.iconDark = tournamentData.icondark
+			end
+
+			-- Manual override
+			local prefix = 'qualifies' .. index
+			data.title = context[prefix .. 'name'] or data.title
+
+			return data
+		end,
+		row = 'qualified',
+		rowParse = function (prizePool, input, context, index)
+			return Logic.readBool(input)
+		end
+	},
+	[prizeTypes.POINTS] = {
+		header = 'points',
+		headerParse = function (prizePool, input, context, index)
+			local pointsData = mw.loadData('Module:Points/data')
+			return pointsData[input] or {title = 'Points'}
+		end,
```

Comment:
some points have abbr stuff in their headers while others do not, do we have to consider something for that?

Reply:
The Display will use the abbr (`titleLong`) is given